docs/reference/data-streams/data-streams.asciidoc

```diff
@@ -47,16 +46,152 @@ We recommend using data streams if you:
 
 
 [discrete]
-[[data-streams-toc]]
-== In this section
+[[backing-indices]]
+== Backing indices
+A data stream consists of one or more _backing indices_. Backing indices are
+<<index-hidden,hidden>>, auto-generated indices used to store a stream's
+documents.
 
-* <<data-streams-overview>>
-* <<set-up-a-data-stream>>
-* <<use-a-data-stream>>
-* <<data-streams-change-mappings-and-settings>>
+image::images/data-streams/data-streams-diagram.svg[align="center"]
 
+To create backing indices, each data stream requires a matching
+<<indices-templates,index template>>. This template acts as a blueprint for the
+stream's backing indices. It contains:
+
+* The mappings and settings applied to each backing index when it's created.
+
+* A name or wildcard (`*`) pattern that matches the data stream's name.
+
+* A `data_stream` object with an empty body (`{ }`). This object indicates the
+template is used for data streams.
+
+A `@timestamp` field must be included in every document indexed to the data
+stream. This field can be mapped as a <<date,`date`>> or
+<<date_nanos,`date_nanos`>> field data type in the stream's matching index
+template. If no mapping is specified in the template, the `date` field data type
+with default options is used.
+
+The same index template can be used to create multiple data streams.
+
+
+[discrete]
+[[data-streams-generation]]
+== Generation
+
+Each data stream tracks its _generation_: a six-digit, zero-padded integer
+that acts as a cumulative count of the data stream's backing indices. This count
+includes any deleted indices for the stream. The generation is incremented
+whenever a new backing index is added to the stream.
+
+When a backing index is created, the index is named using the following
+convention:
+
+[source,text]
+----
+.ds-<data-stream>-<generation>
+----
+
+For example, the `web_server_logs` data stream has a generation of `34`. The
+most recently created backing index for this data stream is named
+`.ds-web_server_logs-000034`.
+
+Because the generation increments with each new backing index, backing indices
+with a higher generation contain more recent data. Backing indices with a lower
+generation contain older data.
+
+A backing index's name can change after its creation due to a
+<<indices-shrink-index,shrink>>, <<snapshots-restore-snapshot,restore>>, or
+other operations.
+
+[discrete]
+[[data-stream-write-index]]
+== Write index
+
+When a read request is sent to a data stream, it routes the request to all its
+backing indices. For example, a search request sent to a data stream would query
+all its backing indices.
+
+image::images/data-streams/data-streams-search-request.svg[align="center"]
+
+However, the most recently created backing index is the data stream’s only
```

Comment:
I would lead with this part and move the part about federating the search across all of the backing indices afterward. Otherwise, it's a bit jarring to start reading about read requests in a section about write requests.